src/Symfony/Bundle/SecurityBundle/Tests/DependencyInjection/SecurityExtensionTest.php

```diff
@@ -123,82 +123,6 @@ public function testDisableRoleHierarchyVoter()
         $this->assertFalse($container->hasDefinition('security.access.role_hierarchy_voter'));
     }
 
-    /**
-     * @group legacy
-     * @expectedDeprecation Setting logout_on_user_change to false is deprecated as of 3.4 and will always be true in 4.0. Set logout_on_user_change to true in your firewall configuration.
-     */
-    public function testDeprecationForUserLogout()
-    {
-        $container = $this->getRawContainer();
-
-        $container->loadFromExtension('security', array(
-            'providers' => array(
-                'default' => array('id' => 'foo'),
-            ),
-
-            'firewalls' => array(
-                'some_firewall' => array(
-                    'pattern' => '/.*',
-                    'http_basic' => null,
-                    'logout_on_user_change' => false,
-                ),
-            ),
-        ));
-
-        $container->compile();
-    }
-
-    /**
-     * @group legacy
-     * @expectedDeprecation Firewall "some_firewall" is configured as "stateless" but the "switch_user.stateless" key is set to false. Both should have the same value, the firewall's "stateless" value will be used as default value for the "switch_user.stateless" key in 4.0.
-     */
-    public function testSwitchUserNotStatelessOnStatelessFirewall()
```

Comment:
I suggest to rename this test, remove the legacy flag and add the following assertion

```php
$this->assertTrue($container->getDefinition('security.authentication.switchuser_listener.default')->getArgument(9));
```